app/views/shared/_hidden_previous_fields.html.erb

```diff
@@ -0,0 +1,7 @@
+<%= form.hidden_field("prev_l", value: params["prev_l"] || params["l"] || "none") %>
```

Comment:
why do we have to assign `none`? can't we skip it if it's `nil`? it'll save you from checking for `none` later?

Reply:
I think I had to do that as if they aren't set on the original request but then the invalid form causes the non-prev parameters to be present in the querystring on the next request, the partial will populate the `prev_` value from the new invalid parameters. `none` is significant as it indicates that there was no parameter in the last valid filter from the results page and stops it being set subsequently.